DataConnectors/Netskope/AzureFunctionNetskope/run.ps1

```diff
@@ -0,0 +1,198 @@
+    $obfurl = $url -replace "token=[a-z0-9]+\&", "token=<apiToken>&"
+    Write-Host "Retrieving '$LogType' events from $obfurl"
+    $response = Invoke-RestMethod -Uri $url
+    if ($response.status -eq "error") {
+        $errorCode = $response.errorCode
+        $errors = $response.errors
+        Write-Host "ERROR encountered while retrieving '$LogType' events - $errorCode - $errors"
+    }
+    else {
+       return $response      
+    }
+}
+
+# Function to retrieve the checkpoint start time of the last successful API call for a given logtype. Checkpoint file will be created if none exists
+function GetStartTime($CheckpointFile, $LogType, $TimeInterval){
+    $firstEndTimeRecord = (Get-Date -Date ((Get-Date).DateTime) -UFormat %s)
+    $firstStartTimeRecord = $firstEndTimeRecord - $TimeInterval
+    if ([System.IO.File]::Exists($CheckpointFile) -eq $false) {
+        $CheckpointLog = @{}
+        foreach ($apiType in $apitypes){
+            $CheckpointLog.Add($apiType,$firstStartTimeRecord)
+        }
+        $CheckpointLog.GetEnumerator() | Select-Object -Property Key,Value | Export-CSV -Path $CheckpointFile -NoTypeInformation
+        return $firstStartTimeRecord 
+    }
+    else{
+        $GetLastRecordTime = Import-Csv -Path $CheckpointFile
+        $startTime = $GetLastRecordTime | ForEach-Object{ 
+                        if($_.Key -eq $LogType){
+                            $_.Value
+                        }
+                    }
+        return $startTime
+    }
+}
+
+# Function to update the checkpoint time with the last successful API call end time
+function UpdateCheckpointTime($CheckpointFile, $LogType, $LastSuccessfulTime){
+    $checkpoints = Import-Csv -Path $CheckpointFile
+    $checkpoints | ForEach-Object{ if($_.Key -eq $LogType){$_.Value = $LastSuccessfulTime}}
+    $checkpoints.GetEnumerator() | Select-Object -Property Key,Value | Export-CSV -Path $CheckpointFile -NoTypeInformation
+}
+
+# Main Function to call the API and Post the response to the Log Analytics API
+function Netskope () {
+
+    $customerId = $env:workspaceId
+    $sharedKey = $env:workspacekey
+    $apikey = $env:apikey
+    $uri = $env:uri
+    $tableName = "Netskope"
+    $timeInterval = [int]($env:timeInterval) * 60
+    $pageLimit = 5000
+    $skip = 0
+    $loggingOptions = $env:logTypes
+    $apitypes = @($loggingOptions.split(",").Trim())
+    $checkPointFile = ".\NetskopeCheckpoint.csv"
+            
+    foreach($logtype in $apitypes){
+
+        $endTime = (Get-Date -Date ((Get-Date).DateTime) -UFormat %s)  
+        $startTime = GetStartTime -CheckpointFile $checkPointFile -LogType $logtype -TimeInterval $timeInterval # function to create starttime
+
+        $alleventobjs = @()
+        $count = 0
+        Do {
+            $response = GetLogs -Uri $uri -ApiKey $apikey -StartTime $startTime -EndTime $endTime -LogType $logtype -Page $pageLimit                         
```

Comment:
Related to previous comment, $skip needs to pass-in to GetLogs as a param

Reply:
this isn't necessary, but I have added for code comprehensibility